src/swbf_fnv_hashes.cpp

```diff
@@ -2819,16 +3868,59 @@ const std::unordered_map<std::uint32_t, std::string_view> swbf_hashes{
    "ZoomRate"_fnvp,
    "ZoomTurnDivisorMax"_fnvp,
    "ZoomTurnDivisorMin"_fnvp,
-   "ZOrder"_fnvp};
+   "ZOrder"_fnvp,
+};
+
+std::mutex swbf_hashes_mutex;
+
 }
 
 std::string lookup_fnv_hash(const std::uint32_t hash)
 {
+   std::string ret_val;
+
+   std::lock_guard lock{swbf_hashes_mutex};
    const auto result = swbf_hashes.find(hash);
 
-   if (result != std::cend(swbf_hashes)) return std::string{result->second};
+   if (result != std::cend(swbf_hashes)) {
+      // this can be interesting when building a dict
+      // synced_cout::print("looked_up:", result->second, "\n");
+      ret_val = std::string{result->second};
+   }
+
+   if (ret_val.empty()) {
+      std::string hex_string = to_hexstring(hash);
+
+      synced_cout::print("Warning: Unknown hash looked up.\n"s, "   value: "s, hex_string,
+                         '\n');
+      ret_val = hex_string;
```

Comment:
We can save a copy here by just putting the hex string directly into `ret_val` then printing from that. Admittedly the string returned by `to_hexstring` is likely to always fit in the small string buffer so eliminating the copy doesn't save much but it also makes the code more concise without really harming readability (unless you think it does?) so may as well.
```suggestion
      ret_val = to_hexstring(hash);

      synced_cout::print("Warning: Unknown hash looked up.\n"s, "   value: "s, ret_val,
                         '\n');
```